src/main/java/net/sf/jabref/logic/integrity/IntegrityCheck.java

```diff
@@ -321,6 +323,12 @@ private BracketChecker(String field) {
                 return Collections.singletonList(new IntegrityMessage(Localization.lang("should contain a four digit number"), entry, FieldName.YEAR));
             }
 
+            if (!ENDS_WITH_FOUR_DIGIT.test(value.get().replaceAll("[](){},.;!?<>%&§]", ""))) {
```

Comment:
Some regex chars have a special function, they are ommited. `String.replaceAll(String regex, String replacement)` I failed to define a variable.